### DISCUSSION TITLE
How to add a button to trigger an AJAX call

### DISCUSSION THREAD

**Discussion by fred-the-coder** [joomengine/Joomla-Component-Builder#867] [Q&A] [answered]
Hi all,

Using a JCB component, I would like to add a button in my edit frontend view that triggers an AJAX call.
I watched the tutorials (AJX and Button) but did not find the how to...

Answers could be:

- how to add a button within edit form? (not in the toolbar, but somewhere in the form)
- how to set this button triggering the AJAX call?

Maybe I have to do it manually by modifying the edit.php and adding custom JS...

Thank you for your sugggestions...

**Comment by aradwan-mcg** [comment, marked as answer]
you can add custom js and php (ajax) via jcb by going to admin views, and you can find the tabs regarding javascript and php, and the button can be created via normal field creation
(note the ajax methods are all stored in the ajax.php model and called via the ajax.json.php controller, so you have to include the controller name in the task like this: task=ajax.methodname

**Reply by fred-the-coder** [reply]
Hi @aradwan-mcg thanks for you reply.
Could you precise which kind of specific JCB field is a "button" ?
Thanks

**Reply by aradwan-mcg** [reply]
there isn't one (i confused it with another matter sorry), you could dynamically create one via js, and assign a click handler to prevent the form from submitting, and then make your ajax call

**Reply by fred-the-coder** [reply]
@aradwan-mcg  let me try!, thanks
Actuallay, I was expecting if JCB was handling this trough it's UI... creatin a form field of button kind and assign it a JS call...

**Reply by Llewellynvdm** [reply]
JCB front-end is very much "you do the work" this was a design choice I made, Since I did not want my clients to feel like I am just copy pasting old projects across for them, and then asking them money... I wanted to ensure that each project has a unique look and feel in the front-end. This resulted in me choosing a very open and blank page approach, where you can paint whatever you like.... 

Many have asked me over the years to change this, but I am not that willing really. Here is the thing, JCB is a wire-framing tool that allow you into the details, that means its a tool for comand-line advance programmers who know how to build components without JCB... so that JCB just speeds-up their tasks. So back-end we do much of the work. But front-end we only give you the table to set your own custom look. This means you need to know JS, CSS, and basic PHP to be able to really enjoy the front-end building ability of JCB.

What is true is that we are talking about adding a few boilerplate options for common task like a list view, grid-view and a few more, so you can more quickly get the idea of how it works... and then tweak to your harts desire.

**Reply by fred-the-coder** [reply]
Hi @Llewellynvdm,
Thanks for taking time to have a look to my request...

I understand your point of view and knew it, since I am using JCB for 2 years. Let me thank you and congrat's again for this great Joomla component...cheery on the cake, accessible for free...

Notice that "you do the work" approach is fine for me as I love coding...but the limitation is regarding ehancement of my "JCB made" Joomla components. Indeed, when I "you do the work", honestly it's outside JCB. Thus I break the link between JCB and the code going live in production. When my customers ask me for enhancements, I have to go back to JCB to create the enhancement. I have to "you do the work" again by comparing the new code generated with JCB and the one in production...which you should know is not as efficient as it should be.
Even if I know JCB provide a lot of customisation hook....
I anticpate your feeling! "OK Fred, thus enhance JCB yourself!" 
Cheers

**Reply by Llewellynvdm** [reply]
Your welcome!

Well actually JCB has round trip development, do you know what that means.... I am not always sure how much others experiment with JCB's features.... So search for that key word and check out all videos... its a powerful.

Next I would encourage you to use GIT... this will help see what JCB changes with new releases. All my projects also live in a GIT repository so that I can monitor the changes as I compile them with new JCB versions.

Then there is the [new powers of JCB](https://youtu.be/GLcPdLXQxrQ) a whole new world that has become my main fun place... custom code use to be that place, but this new power area is just liberating, and combine that with round trip development and you have super powers... lol
